test/object-store/util/test_utils.hpp

```diff
@@ -102,6 +169,24 @@ std::string get_parent_directory(const std::string& path);
 
 #define REQUIRE_THROWS_CONTAINING(expr, msg) REQUIRE_THROWS_WITH(expr, Catch::Matchers::ContainsSubstring(msg))
 
+namespace {
+}
+
+#define REQUIRE_EXCEPTION(expr, c, msg)                                                                              \
+    REQUIRE_THROWS_MATCHES(expr, realm::Exception, _impl::make_exception_matcher(realm::ErrorCodes::c, msg))
+#define REQUIRE_THROWS_OUT_OF_BOUNDS(expr, index, size, msg)                                                         \
+    REQUIRE_THROWS_MATCHES(expr, OutOfBounds, OutOfBoundsMatcher(index, size, msg));
+
+#define REQUIRE_THROWS_WITH_CODE(expr, err)                                                                          \
+    do {                                                                                                             \
+        try {                                                                                                        \
+            expr;                                                                                                    \
+        }                                                                                                            \
+        catch (const Exception& e) {                                                                                 \
+            REQUIRE(e.code() == err);                                                                                \
+        }                                                                                                            \
+    } while (0)
```

Comment:
This all looks like a bad merge or something? REQUIRE_THROWS_WITH_CODE is broken (it passes if the expression doesn't throw) and is no longer used, and there's an empty namespace.

Reply:
Hmm, the file is taken directly from your tg/exception-unification branch, so not merged in any way. Anyway - I have now fixed REQUIRE_THROW_LOGIC_ERROR_WITH_CODE to fail when not throwing.

Reply:
This code is not present in the commits in my branch which you pulled into this PR.